auth/auth0_manager.go

```diff
@@ -36,16 +41,32 @@ const (
 	validAuthExpiry = 1 * time.Hour
 )
 
-func NewAuth0Manager(domain, clientID, connection string) *Auth0Manager {
+func NewAuth0Manager(domain, clientID, connection string, throttleMin int) *Auth0Manager {
```

Comment:
This seems over-complicated. 

I think the `Auth0Manager` only needs one more field, `throttle <-chan Time`. 
We should take the rateLimit as a `time.Duration` param, and just generate the throttle field using only that rate, e.g:
```
func NewAuth0Manager(domain, clientID, connection string, rateLimit time.Duration) *Auth0Manager {
    return &Auth0Manager{
        ...
        throttle: time.Tick(rateLimit),
    }
}
```

That way in tests, we can just pass in a really high rate limit.